### DISCUSSION TITLE
SkyBlock Inventory "type"

### DISCUSSION THREAD

**Discussion by TheNullicorn** [HypixelDev/PublicAPI#311] [Q&A]
Hello! I posted this on the forums the other day but was hoping I could get a better answer here.

In the SkyBlock API, every inventory/item-related field is stored as an object with a `data` and `type` field. As far as I've seen, every instance of this always has type set to `0`. What exactly does `type` indicate, and is it ever anything other than `0`?

The old [pre-1.8 window IDs](https://wiki.vg/index.php?title=Inventory&oldid=6002#Windows) come to mind (where a chest would have the ID `0`), but that wouldn't make much sense if Hypixel runs a 1.8 server. Someone on my forum thread linked to a [Thirtyvirus video](https://www.youtube.com/watch?v=Laz4cOm5RCw&t=14m17s) that says that Hypixel actually runs a modified 1.7 server, but I've never heard that before and was wondering how true that was.

Thanks!

**Comment by IamMusavaRibica** [comment]
I'd say Hypixel runs on a modified 1.7 servers, because, vanilla commands that got added in 1.8 don't work (Unknown command, even gets translated if your language is different), but I don't know how would the server handle items like granite, diorite, andesite, slime blocks, wet sponge, armor stands, banners etc.

In Spigot, I'm sure you can change the amount of rows you see when a chest-type container is opened (can be from 1 to 6). Since most, if not all of the Skyblock menus / GUIs have the inventories just like chests do, but filled with black stained glass (also called menu glass) I'd say the type set to 0 is correct, because the inventory is chest-type.
